### PR TITLE
src/components: add component TableNotifications for displaying user notifications

### PR DESCRIPTION
**Pull request by maceksimon**
Add component `TableNotifications` for displaying user notifications on Profile page.

* Add Quasar table with custom slots.
* Add `Notification` type based on [Notification API endpoint](https://dpnk.dopracenakole.cz/rest/notification/).
* Add `TableColumn` type to emulate Quasar table prop types and prevent typescript error.
* Add translations
* Add tests

### CODE REVIEW

File: src/components/profile/TableNotifications.vue [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+              :href="props.row.data.url"
+              :class="props.row.unread ? 'text-weight-bold' : ''"
+              target="_blank"
+              data-cy="notification-verbal"
+              >{{ props.row.verb }}</component
+            >
+          </q-td>
+          <q-td key="timestamp" :props="props" data-cy="notification-timestamp">
+            <template v-for="col in props.cols" :key="col.field">
+              <span v-if="col.field === 'timestamp'">
```
Comment (tmszi) [inline src/components/profile/TableNotifications.vue:149]: You can use TS Enum for `'timestamp'` literal.

File: src/components/profile/TableNotifications.vue [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+// types
+import type { TableColumn } from '../types/Table';
+import type { Notification } from '../types/Notifications';
+
+export default defineComponent({
+  name: 'TableNotifications',
+  setup() {
+    const columns: TableColumn[] = [
+      {
+        name: 'title',
```
Comment (tmszi) [inline src/components/profile/TableNotifications.vue:37]: `name` key value string literal should be TS Enum.

**Comment by tmszi** [comment]
First table row Notification column with text _New message_ is not behave as URL link, isn't that a bug?

File: src/components/profile/TableNotifications.vue (hunk 1 above)
Comment (maceksimon) [reply src/components/profile/TableNotifications.vue]: Refactored into enum.

File: src/components/profile/TableNotifications.vue (hunk 2 above)
Comment (maceksimon) [reply src/components/profile/TableNotifications.vue]: Refactored into enum.

**Comment by maceksimon** [comment]
> If I click on the table column Notification row link e.g. _Challenge completed_, should not be row marked as readed?

Yes, this is how it should work.
I implemented a onNotificationClick function that handles this basic case.

**Comment by maceksimon** [comment]
> First table row Notification column with text _New message_ is not behave as URL link, isn't that a bug?

This is intended. In `tableNotifications.json`, the url is intentionally left out and so it displays title without a link.
Not sure if this can happen in production, but this way, the UI supports "non-clickable" notification if we need it.

**Review by tmszi — APPROVED** [review]

**Pull request merged**